### PR TITLE
Degree search sort

### PR DESCRIPTION
**Pull request by jmbarne3**
Sort degrees based on program_type and add total number of results.

### CODE REVIEW

File: dev/custom-pages/homepage.scss [hunk 1]
```diff
@@ -270,8 +269,29 @@ $color-linkedin: #0077b5;
...
+      .padded-top {
+        padding-top: 95px;
+      }
+
+      .padded-bottom {
+        padding-bottom: 0;
+      }
+    }
+
+    @media (max-width: $screen-sm-max) {
```
Comment (rjucf) [inline dev/custom-pages/homepage.scss]: This should be the default css for a mobile first approach.

File: src/js/script.js [hunk 2]
```diff
@@ -1483,30 +1483,64 @@ var academicDegreeSearch = function ($) {
  * #search-query specific typeahead init, event handlers
  **/
   var $academicsDegreeSearch = $('#academics-degree-search');
+  var resultCount = 0;
```
Comment (rjucf) [inline src/js/script.js]: Minor: These vars could be combined

File: dev/custom-pages/homepage.scss (hunk 1 above)
Comment (rjucf) [inline dev/custom-pages/homepage.scss]: Nevermind, that is old code I left in instead of fixing.

File: dev/custom-pages/homepage.scss [hunk 3]
```diff
@@ -270,8 +269,29 @@ $color-linkedin: #0077b5;
     }
 
     .padded-bottom {
-      padding-bottom: 70px;
+      padding-bottom: 47px;
     }
+
+    @media (max-width: $screen-md-max) {
```
Comment (cjg89) [inline dev/custom-pages/homepage.scss]: Avoid nested selectors within media queries when using sass--something like
```
img {
  @media (max-width: $screen-md-max) {
    top: 73px;
  }
}
```
is preferred (per-selector media queries).

File: src/js/script.js [hunk 4]
```diff
@@ -1482,31 +1482,65 @@ var academicDegreeSearch = function ($) {
...
 
-    var degreesWithDefaults = function (q, sync) {
-      if (q === '') {
-        sync(degrees.get('Bachelor', 'Graduate', 'Certificate'));
-      } else {
-        degrees.search(q, sync);
-      }
-    };
+    $('.degree-search-box, .academics-search-box').on('click', '#show-all-degrees', function () {
+      window.location = '/degree-search/?search-query=' + $('#academics-degree-search').val();
```
Comment (cjg89) [inline src/js/script.js]: Does window.location navigate to the correct page regardless of where it's called from?  (e.g. clicks from the home page and academics are the same)?

File: src/js/script.js [hunk 5]
```diff
@@ -1526,7 +1570,9 @@ var academicDegreeSearch = function ($) {
             'No degrees found for search term.',
             '</div>' + degreeType
           ].join('\n'),
-          footer: degreeType
+          footer: function (context) {
+            return '<div class="tt-suggestion tt-selectable"><a href="#" id="show-all-degrees">Show all ' + resultCount + ' degress for <em>&ldquo;' + context.query + '&rdquo;</em></a></div>' + degreeType;
```
Comment (cjg89) [inline src/js/script.js]: Typo "show all _degress_"

File: dev/custom-pages/homepage.scss (hunk 3 above)
Comment (jmbarne3) [inline dev/custom-pages/homepage.scss]: I'm fairly certain these css changes shouldn't even be in here... They're completely unrelated to the change set. I'm not quite sure how they got in here.

File: src/js/script.js (hunk 4 above)
Comment (jmbarne3) [inline src/js/script.js]: In dev, it is loading the correct page in both cases. window.location treats a `/url.html` as relative to the root, not the current path.

File: dev/custom-pages/homepage.scss (hunk 3 above)
Comment (jmbarne3) [inline dev/custom-pages/homepage.scss]: Changes were not unrelated and are being kept. Updated to have each rule have the media query within it.

**Pull request merged**